agent/bench.py

```diff
@@ -486,9 +487,11 @@ def generate_nginx_config(self):
             "code_server": codeserver,
         }
         nginx_config = os.path.join(self.directory, "nginx.conf")
-        self.server._render_template(
-            "bench/nginx.conf.jinja2", config, nginx_config
-        )
+
+        with FileLock(nginx_config + ".lock"):
```

Comment:
@balamurali27 This is incorrect. 

If there are two processes waiting to write config then they will still override one another. The lock should be acquired as first step in the process.